pkg/network/common/egressip.go

```diff
@@ -402,3 +441,100 @@ func (eit *EgressIPTracker) Ping(ip string, timeout time.Duration) bool {
 		return true
 	}
 }
+
+// Finds the best node to allocate the egress IP to, given the existing allocation. The
+// boolean return value indicates whether multiple nodes could host the IP.
+func (eit *EgressIPTracker) findEgressIPAllocation(ip net.IP, allocation map[string][]string) (string, bool) {
+	bestNode := ""
+	otherNodes := false
+
+	for _, node := range eit.nodesByNodeIP {
+		egressIPs, exists := allocation[node.nodeName]
+		if !exists {
+			continue
+		}
+		for _, parsed := range node.parsedCIDRs {
+			if parsed.Contains(ip) {
+				if bestNode != "" {
+					otherNodes = true
+					if len(allocation[bestNode]) < len(egressIPs) {
+						continue
```

Comment:
break instead of continue?
Continue may not help because even if other cidr on the node matched the ip, len(egressIPs) for the node won't change and it will be greater than allocation[bestNode].